### PR TITLE
fixes header action selector

### PR DESCRIPTION
**Pull request by kvendrik**
### WHY are these changes introduced?
fixes https://github.com/Shopify/polaris-react-deprecated/issues/2463.
moved from https://github.com/Shopify/polaris-react-deprecated/pull/2487.

### WHAT is this pull request doing?
Replaces the wildcard action selector with an actual selector that points to secondary actions in the page header.

### How to 🎩
1. Add [a Page with both `actionGroups` and `secondaryActions`](https://github.com/Shopify/polaris-react/blob/8952a87b89bc86fffd295dc0316820cdb06fca10/playground/Playground.tsx) to the Playground.
2. Make sure the selector works by inspecting the first and last action.

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [x] Updated [UNRELEASED.md](https://github.com/Shopify/polaris-react/blob/master/UNRELEASED.md)

### CODE REVIEW

**Comment by danrosenthal** [comment]
@kvendrik is this still a change you'd like to see merged? I'd recommend rebasing and re-pinging for review if so. Otherwise it should be closed.

**Comment by kvendrik** [comment]
@danrosenthal this is ready to be reviewed, tagged @solonaarmstrong last week but looks like she hasn't had time yet. Will rebase after review.

File: src/components/Page/components/Header/Header.tsx [hunk 1]
```diff
@@ -184,16 +184,17 @@ export default class Header extends React.PureComponent<Props, State> {
...
-              key={title}
-              title={title}
-              icon={icon}
-              actions={actions}
-              details={details}
-              onOpen={this.handleActionGroupOpen}
-              onClose={this.handleActionGroupClose}
-              active={title === openActionGroup}
-            />
+            <div className={styles.IndividualAction} key={title}>
```
Comment (chloerice) [inline src/components/Page/components/Header/Header.tsx]: Should this have a more unique key similar to the one given to secondary actions on line 270?

```suggestion
            <div
              className={styles.IndividualAction}
              key={`ActionGroup-${title}-${index}`}
            >
```

File: UNRELEASED.md [hunk 2]
```diff
@@ -6,6 +6,8 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 
 ### Enhancements
 
+- Fixes `Action`s selector in `Page`'s `Header` component ([#2487](https://github.com/Shopify/polaris-react/pull/2487))
```
Comment (solonaarmstrong) [inline UNRELEASED.md]: Fixed

File: src/components/Page/components/Header/Header.tsx [hunk 3]
```diff
@@ -183,17 +183,21 @@ export default class Header extends React.PureComponent<Props, State> {
...
+              key={`ActionGroup-${title}-${index}`}
+            >
+              <ActionGroup
+                title={title}
+                icon={icon}
+                actions={actions}
+                details={details}
+                onOpen={this.handleActionGroupOpen}
+                onClose={this.handleActionGroupClose}
+                active={title === openActionGroup}
```
Comment (solonaarmstrong) [inline src/components/Page/components/Header/Header.tsx]: Should this logic be in a variable?

**Pull request merged**